test/jasmine/tests/transform_groupby_test.js

```diff
@@ -9,6 +9,15 @@ var customAssertions = require('../assets/custom_assertions');
 var assertDims = customAssertions.assertDims;
 var assertStyle = customAssertions.assertStyle;
 
+
+function supplyDataDefaults(dataIn, dataOut) {
+    return Plots.supplyDataDefaults(dataIn, dataOut, {}, {
```

Comment:
It's getting more difficult to call anything but `Plots.supplyDefaults` in tests due to the large amount of things to mocks. We should maybe start standardising around [`assets/supply_defaults.js`](https://github.com/plotly/plotly.js/blob/master/test/jasmine/assets/supply_defaults.js) instead?

Reply:
right, that's the disadvantage of this kind of precalculation, which I've added a lot of lately 🙈

👍  for moving toward `assets/supply_defaults.js`.